### PR TITLE
Feature/add benchmarking

### PR DESCRIPTION
**Pull request by MartinCarpentier**
I added a simple benchmark project, and also renamed the old test project to reflect what it contains.

The result of the feature collection serialize benchmark i ran on my own pc.

``` ini

BenchmarkDotNet=v0.13.1, OS=Windows 10.0.19042.1415 (20H2/October2020Update)
AMD Ryzen 7 3800X, 1 CPU, 16 logical and 8 physical cores
.NET SDK=6.0.100
  [Host]        : .NET Core 3.1.21 (CoreCLR 4.700.21.51404, CoreFX 4.700.21.51508), X64 RyuJIT
  .NET 5.0      : .NET 5.0.12 (5.0.1221.52207), X64 RyuJIT
  .NET 6.0      : .NET 6.0.0 (6.0.21.52210), X64 RyuJIT
  .NET Core 3.1 : .NET Core 3.1.21 (CoreCLR 4.700.21.51404, CoreFX 4.700.21.51508), X64 RyuJIT


```
|                  Method |           Job |       Runtime |       N |       Mean |    Error |   StdDev | Ratio | RatioSD |       Gen 0 |      Gen 1 |    Gen 2 | Allocated |
|------------------------ |-------------- |-------------- |-------- |-----------:|---------:|---------:|------:|--------:|------------:|-----------:|---------:|----------:|
|     **SerializeNewtonsoft** |      **.NET 5.0** |      **.NET 5.0** |  **100000** |   **447.9 ms** |  **3.64 ms** |  **3.40 ms** |  **1.19** |    **0.01** |  **10000.0000** |  **3000.0000** |        **-** |    **112 MB** |
|     SerializeNewtonsoft |      .NET 6.0 |      .NET 6.0 |  100000 |   377.6 ms |  2.34 ms |  1.95 ms |  1.00 |    0.00 |  10000.0000 |  3000.0000 |        - |    112 MB |
|     SerializeNewtonsoft | .NET Core 3.1 | .NET Core 3.1 |  100000 |   473.2 ms |  4.73 ms |  4.19 ms |  1.25 |    0.02 |  10000.0000 |  3000.0000 |        - |    112 MB |
|                         |               |               |         |            |          |          |       |         |             |            |          |           |
| SerializeSystemTextJson |      .NET 5.0 |      .NET 5.0 |  100000 |   233.7 ms |  0.94 ms |  0.88 ms |  1.28 |    0.01 |   1666.6667 |   333.3333 | 333.3333 |     65 MB |
| SerializeSystemTextJson |      .NET 6.0 |      .NET 6.0 |  100000 |   183.2 ms |  1.54 ms |  1.44 ms |  1.00 |    0.00 |   1333.3333 |          - |        - |     35 MB |
| SerializeSystemTextJson | .NET Core 3.1 | .NET Core 3.1 |  100000 |   244.9 ms |  4.46 ms |  4.17 ms |  1.34 |    0.02 |   1500.0000 |          - |        - |     65 MB |
|                         |               |               |         |            |          |          |       |         |             |            |          |           |
|     **SerializeNewtonsoft** |      **.NET 5.0** |      **.NET 5.0** | **1000000** | **4,589.9 ms** | **15.37 ms** | **14.38 ms** |  **1.19** |    **0.00** | **109000.0000** | **27000.0000** |        **-** |  **1,123 MB** |
|     SerializeNewtonsoft |      .NET 6.0 |      .NET 6.0 | 1000000 | 3,846.8 ms |  7.69 ms |  7.19 ms |  1.00 |    0.00 | 109000.0000 | 27000.0000 |        - |  1,123 MB |
|     SerializeNewtonsoft | .NET Core 3.1 | .NET Core 3.1 | 1000000 | 4,870.0 ms | 56.60 ms | 50.18 ms |  1.27 |    0.01 | 109000.0000 | 27000.0000 |        - |  1,123 MB |
|                         |               |               |         |            |          |          |       |         |             |            |          |           |
| SerializeSystemTextJson |      .NET 5.0 |      .NET 5.0 | 1000000 | 2,113.3 ms |  4.23 ms |  3.75 ms |  1.17 |    0.00 |  16000.0000 |          - |        - |    605 MB |
| SerializeSystemTextJson |      .NET 6.0 |      .NET 6.0 | 1000000 | 1,809.1 ms |  4.25 ms |  3.98 ms |  1.00 |    0.00 |  16000.0000 |          - |        - |    351 MB |
| SerializeSystemTextJson | .NET Core 3.1 | .NET Core 3.1 | 1000000 | 2,581.5 ms | 41.94 ms | 35.02 ms |  1.43 |    0.02 |  16000.0000 |          - |        - |    605 MB |

The memory footprint seems to be alot smaller with System.Text.Json.
Especially with .NET 6.

### CODE REVIEW

File: src/GeoJSON.Text.Test.Benchmark/Deserialize/JsonEmbeddedFileReader.cs [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+            var myType = typeof(JsonEmbeddedFileReader);
+            var currentNamespace = myType.Namespace;
+
+            var assembly = Assembly.GetCallingAssembly();
+
+            var fileStreamName = $"{currentNamespace}.{fileName}.json";
+
+            Console.WriteLine(fileStreamName);
+
+            using (Stream stream = assembly.GetManifestResourceStream(fileStreamName))
```
Comment (matt-lethargic) [inline src/GeoJSON.Text.Test.Benchmark/Deserialize/JsonEmbeddedFileReader.cs:20]: Should ideally aim for no warnings in build

File: src/GeoJSON.Text.Test.Benchmark/Deserialize/DeserializeFeatureLinestring.cs [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+    {
+        string fileContents = "";
+
+        [GlobalSetup]
+        public void Setup()
+        {
+            fileContents = JsonEmbeddedFileReader.GetExpectedJson("DeserializeFeatureLinestring");
+        }
+
+#pragma warning disable CS8603 // Possible null reference return.
```
Comment (matt-lethargic) [inline src/GeoJSON.Text.Test.Benchmark/Deserialize/DeserializeFeatureLinestring.cs:20]: If we're going to supress the null checks for this project would it be easier to set at project level? `<nullable>disable</nullable>`

**Review by matt-lethargic — APPROVED** [review]
This is brilliant, we thought years ago about getting some benchmarks into GeoJSON.Net but never had the appetite as we didn't have any more features to add. Couple of comments about nulls but nothing that would stop the PR.

Could it be an idea to use this [Continuous Benchmark](https://github.com/marketplace/actions/continuous-benchmark-net)?

Sorry I just completed the last one, I had some time spare while sat at computer and just thought I could do as was only a couple of little changes.

File: src/GeoJSON.Text.Test.Benchmark/Deserialize/DeserializeFeatureLinestring.cs (hunk 2 above)
Comment (MartinCarpentier) [reply src/GeoJSON.Text.Test.Benchmark/Deserialize/DeserializeFeatureLinestring.cs]: Now throws NullReferenceException if result is null.

File: src/GeoJSON.Text.Test.Benchmark/Deserialize/JsonEmbeddedFileReader.cs (hunk 1 above)
Comment (MartinCarpentier) [reply src/GeoJSON.Text.Test.Benchmark/Deserialize/JsonEmbeddedFileReader.cs]: Now throws NullReferenceException if stream is null.

**Comment by MartinCarpentier** [comment]
I will attempt to incorporate Continous Benchmark, but we should be a bit vary of the results when the benchmarking is running on shared runners, and not on a fixed hardware machine setup.

**Comment by MartinCarpentier** [comment]
Ran all the benchmarks, and this is the result for my pc.

``` ini

BenchmarkDotNet=v0.13.1, OS=Windows 10.0.19042.1415 (20H2/October2020Update)
Intel Core i7-9850H CPU 2.60GHz, 1 CPU, 12 logical and 6 physical cores
.NET SDK=6.0.100
  [Host]     : .NET Core 3.1.22 (CoreCLR 4.700.21.56803, CoreFX 4.700.21.57101), X64 RyuJIT
  Job-YLUSEG : .NET 6.0.0 (6.0.21.52210), X64 RyuJIT
  Job-XJPUIK : .NET 5.0.13 (5.0.1321.56516), X64 RyuJIT
  Job-UUWPJS : .NET Core 3.1.22 (CoreCLR 4.700.21.56803, CoreFX 4.700.21.57101), X64 RyuJIT

IterationCount=3  LaunchCount=1  WarmupCount=3  

```
|                  Method |        Job |       Runtime |     Mean |     Error |    StdDev | Rank |  Gen 0 | Allocated |
|------------------------ |----------- |-------------- |---------:|----------:|----------:|-----:|-------:|----------:|
| SerializeSystemTextJson | Job-YLUSEG |      .NET 6.0 | 3.541 μs | 0.4421 μs | 0.0242 μs |    I | 0.2060 |      1 KB |
|                         |            |               |          |           |           |      |        |           |
| SerializeSystemTextJson | Job-XJPUIK |      .NET 5.0 | 3.949 μs | 2.0223 μs | 0.1108 μs |    I | 0.2060 |      1 KB |
|                         |            |               |          |           |           |      |        |           |
| SerializeSystemTextJson | Job-UUWPJS | .NET Core 3.1 | 4.441 μs | 2.2368 μs | 0.1226 μs |    I | 0.2060 |      1 KB |
|                         |            |               |          |           |           |      |        |           |
|     SerializeNewtonsoft | Job-YLUSEG |      .NET 6.0 | 5.362 μs | 3.1303 μs | 0.1716 μs |   II | 0.6104 |      4 KB |
|                         |            |               |          |           |           |      |        |           |
|     SerializeNewtonsoft | Job-XJPUIK |      .NET 5.0 | 5.818 μs | 2.7505 μs | 0.1508 μs |   II | 0.6104 |      4 KB |
|                         |            |               |          |           |           |      |        |           |
|     SerializeNewtonsoft | Job-UUWPJS | .NET Core 3.1 | 6.431 μs | 1.1545 μs | 0.0633 μs |   II | 0.6104 |      4 KB |

``` ini

BenchmarkDotNet=v0.13.1, OS=Windows 10.0.19042.1415 (20H2/October2020Update)
Intel Core i7-9850H CPU 2.60GHz, 1 CPU, 12 logical and 6 physical cores
.NET SDK=6.0.100
  [Host]     : .NET Core 3.1.22 (CoreCLR 4.700.21.56803, CoreFX 4.700.21.57101), X64 RyuJIT
  Job-YLUSEG : .NET 6.0.0 (6.0.21.52210), X64 RyuJIT
  Job-UUWPJS : .NET Core 3.1.22 (CoreCLR 4.700.21.56803, CoreFX 4.700.21.57101), X64 RyuJIT
  Job-XJPUIK : .NET 5.0.13 (5.0.1321.56516), X64 RyuJIT

IterationCount=3  LaunchCount=1  WarmupCount=3  

```
|                  Method |        Job |       Runtime |     N |      Mean |      Error |    StdDev | Rank |     Gen 0 |    Gen 1 |    Gen 2 | Allocated |
|------------------------ |----------- |-------------- |------ |----------:|-----------:|----------:|-----:|----------:|---------:|---------:|----------:|
| SerializeSystemTextJson | Job-YLUSEG |      .NET 6.0 |  1000 |  1.657 ms |  0.2672 ms | 0.0146 ms |    I |   70.3125 |  70.3125 |  70.3125 |    360 KB |
|                         |            |               |       |           |            |           |      |           |          |          |           |
| SerializeSystemTextJson | Job-UUWPJS | .NET Core 3.1 |  1000 |  2.088 ms |  0.6223 ms | 0.0341 ms |    I |   70.3125 |  70.3125 |  70.3125 |    360 KB |
|                         |            |               |       |           |            |           |      |           |          |          |           |
| SerializeSystemTextJson | Job-XJPUIK |      .NET 5.0 |  1000 |  2.090 ms |  2.2827 ms | 0.1251 ms |    I |   70.3125 |  70.3125 |  70.3125 |    360 KB |
|                         |            |               |       |           |            |           |      |           |          |          |           |
|     SerializeNewtonsoft | Job-YLUSEG |      .NET 6.0 |  1000 |  3.251 ms |  1.2135 ms | 0.0665 ms |   II |  164.0625 |  82.0313 |  82.0313 |  1,160 KB |
|                         |            |               |       |           |            |           |      |           |          |          |           |
|     SerializeNewtonsoft | Job-XJPUIK |      .NET 5.0 |  1000 |  3.584 ms |  0.2893 ms | 0.0159 ms |   II |  164.0625 |  82.0313 |  82.0313 |  1,160 KB |
|                         |            |               |       |           |            |           |      |           |          |          |           |
|     SerializeNewtonsoft | Job-UUWPJS | .NET Core 3.1 |  1000 |  3.923 ms |  0.4800 ms | 0.0263 ms |   II |  164.0625 |  78.1250 |  78.1250 |  1,160 KB |
|                         |            |               |       |           |            |           |      |           |          |          |           |
| SerializeSystemTextJson | Job-XJPUIK |      .NET 5.0 | 10000 | 18.295 ms |  7.9428 ms | 0.4354 ms |    I |  500.0000 | 312.5000 | 312.5000 |  5,644 KB |
|                         |            |               |       |           |            |           |      |           |          |          |           |
| SerializeSystemTextJson | Job-YLUSEG |      .NET 6.0 | 10000 | 19.434 ms | 41.6263 ms | 2.2817 ms |    I |  343.7500 | 156.2500 | 156.2500 |  3,595 KB |
|                         |            |               |       |           |            |           |      |           |          |          |           |
| SerializeSystemTextJson | Job-UUWPJS | .NET Core 3.1 | 10000 | 22.558 ms | 24.4346 ms | 1.3393 ms |    I |  531.2500 | 343.7500 | 343.7500 |  5,643 KB |
|                         |            |               |       |           |            |           |      |           |          |          |           |
|     SerializeNewtonsoft | Job-YLUSEG |      .NET 6.0 | 10000 | 33.722 ms | 11.8908 ms | 0.6518 ms |   II | 1687.5000 | 750.0000 | 250.0000 | 11,499 KB |
|                         |            |               |       |           |            |           |      |           |          |          |           |
|     SerializeNewtonsoft | Job-XJPUIK |      .NET 5.0 | 10000 | 38.626 ms |  5.3094 ms | 0.2910 ms |   II | 1714.2857 | 785.7143 | 285.7143 | 11,499 KB |
|                         |            |               |       |           |            |           |      |           |          |          |           |
|     SerializeNewtonsoft | Job-UUWPJS | .NET Core 3.1 | 10000 | 40.075 ms |  7.2834 ms | 0.3992 ms |   II | 1615.3846 | 692.3077 | 230.7692 | 11,499 KB |

``` ini

BenchmarkDotNet=v0.13.1, OS=Windows 10.0.19042.1415 (20H2/October2020Update)
Intel Core i7-9850H CPU 2.60GHz, 1 CPU, 12 logical and 6 physical cores
.NET SDK=6.0.100
  [Host]     : .NET Core 3.1.22 (CoreCLR 4.700.21.56803, CoreFX 4.700.21.57101), X64 RyuJIT
  Job-XJPUIK : .NET 5.0.13 (5.0.1321.56516), X64 RyuJIT
  Job-YLUSEG : .NET 6.0.0 (6.0.21.52210), X64 RyuJIT
  Job-UUWPJS : .NET Core 3.1.22 (CoreCLR 4.700.21.56803, CoreFX 4.700.21.57101), X64 RyuJIT

IterationCount=3  LaunchCount=1  WarmupCount=3  

```
|                    Method |        Job |       Runtime |      Mean |     Error |    StdDev | Rank |  Gen 0 |  Gen 1 | Allocated |
|-------------------------- |----------- |-------------- |----------:|----------:|----------:|-----:|-------:|-------:|----------:|
| DeserializeSystemTextJson | Job-XJPUIK |      .NET 5.0 |  9.358 μs | 0.7175 μs | 0.0393 μs |    I | 0.4272 |      - |      3 KB |
|                           |            |               |           |           |           |      |        |        |           |
| DeserializeSystemTextJson | Job-YLUSEG |      .NET 6.0 |  9.742 μs | 2.9807 μs | 0.1634 μs |    I | 0.4272 |      - |      3 KB |
|                           |            |               |           |           |           |      |        |        |           |
| DeserializeSystemTextJson | Job-UUWPJS | .NET Core 3.1 | 12.850 μs | 0.6955 μs | 0.0381 μs |    I | 0.4578 |      - |      3 KB |
|                           |            |               |           |           |           |      |        |        |           |
|     DeserializeNewtonsoft | Job-YLUSEG |      .NET 6.0 | 14.989 μs | 2.1729 μs | 0.1191 μs |   II | 1.9531 | 0.0458 |     12 KB |
|                           |            |               |           |           |           |      |        |        |           |
|     DeserializeNewtonsoft | Job-XJPUIK |      .NET 5.0 | 15.987 μs | 1.2193 μs | 0.0668 μs |   II | 1.9531 | 0.0305 |     12 KB |
|                           |            |               |           |           |           |      |        |        |           |
|     DeserializeNewtonsoft | Job-UUWPJS | .NET Core 3.1 | 19.406 μs | 0.9241 μs | 0.0507 μs |   II | 1.9531 | 0.0305 |     12 KB |

``` ini

BenchmarkDotNet=v0.13.1, OS=Windows 10.0.19042.1415 (20H2/October2020Update)
Intel Core i7-9850H CPU 2.60GHz, 1 CPU, 12 logical and 6 physical cores
.NET SDK=6.0.100
  [Host]     : .NET Core 3.1.22 (CoreCLR 4.700.21.56803, CoreFX 4.700.21.57101), X64 RyuJIT
  Job-YLUSEG : .NET 6.0.0 (6.0.21.52210), X64 RyuJIT
  Job-XJPUIK : .NET 5.0.13 (5.0.1321.56516), X64 RyuJIT
  Job-UUWPJS : .NET Core 3.1.22 (CoreCLR 4.700.21.56803, CoreFX 4.700.21.57101), X64 RyuJIT

IterationCount=3  LaunchCount=1  WarmupCount=3  

```
|                    Method |        Job |       Runtime |     N |       Mean |       Error |     StdDev | Rank |      Gen 0 |     Gen 1 |    Gen 2 | Allocated |
|-------------------------- |----------- |-------------- |------ |-----------:|------------:|-----------:|-----:|-----------:|----------:|---------:|----------:|
| DeserializeSystemTextJson | Job-YLUSEG |      .NET 6.0 |  1000 |   7.306 ms |   3.4245 ms |  0.1877 ms |    I |   328.1250 |  164.0625 |        - |      2 MB |
|                           |            |               |       |            |             |            |      |            |           |          |           |
| DeserializeSystemTextJson | Job-XJPUIK |      .NET 5.0 |  1000 |   7.608 ms |   0.5361 ms |  0.0294 ms |    I |   328.1250 |  156.2500 |        - |      2 MB |
|                           |            |               |       |            |             |            |      |            |           |          |           |
| DeserializeSystemTextJson | Job-UUWPJS | .NET Core 3.1 |  1000 |   8.924 ms |   7.2798 ms |  0.3990 ms |    I |   343.7500 |  171.8750 |        - |      2 MB |
|                           |            |               |       |            |             |            |      |            |           |          |           |
|     DeserializeNewtonsoft | Job-YLUSEG |      .NET 6.0 |  1000 |  13.572 ms |   1.9034 ms |  0.1043 ms |   II |  1453.1250 |  187.5000 |  78.1250 |      9 MB |
|                           |            |               |       |            |             |            |      |            |           |          |           |
|     DeserializeNewtonsoft | Job-XJPUIK |      .NET 5.0 |  1000 |  14.148 ms |   2.2455 ms |  0.1231 ms |   II |  1453.1250 |  203.1250 |  78.1250 |      9 MB |
|                           |            |               |       |            |             |            |      |            |           |          |           |
|     DeserializeNewtonsoft | Job-UUWPJS | .NET Core 3.1 |  1000 |  14.784 ms |   3.5171 ms |  0.1928 ms |   II |  1437.5000 |  234.3750 |  93.7500 |      9 MB |
|                           |            |               |       |            |             |            |      |            |           |          |           |
| DeserializeSystemTextJson | Job-YLUSEG |      .NET 6.0 | 10000 |  75.983 ms |  19.6159 ms |  1.0752 ms |    I |  3714.2857 | 1714.2857 | 714.2857 |     20 MB |
|                           |            |               |       |            |             |            |      |            |           |          |           |
| DeserializeSystemTextJson | Job-XJPUIK |      .NET 5.0 | 10000 |  83.739 ms |  86.1600 ms |  4.7227 ms |    I |  3714.2857 | 1714.2857 | 714.2857 |     20 MB |
|                           |            |               |       |            |             |            |      |            |           |          |           |
| DeserializeSystemTextJson | Job-UUWPJS | .NET Core 3.1 | 10000 |  92.786 ms |  32.6638 ms |  1.7904 ms |    I |  3600.0000 | 1400.0000 | 600.0000 |     20 MB |
|                           |            |               |       |            |             |            |      |            |           |          |           |
|     DeserializeNewtonsoft | Job-XJPUIK |      .NET 5.0 | 10000 | 141.116 ms |  44.2399 ms |  2.4249 ms |   II | 14000.0000 | 2000.0000 |        - |     87 MB |
|                           |            |               |       |            |             |            |      |            |           |          |           |
|     DeserializeNewtonsoft | Job-YLUSEG |      .NET 6.0 | 10000 | 151.036 ms | 156.1770 ms |  8.5606 ms |   II | 14000.0000 | 2000.0000 |        - |     87 MB |
|                           |            |               |       |            |             |            |      |            |           |          |           |
|     DeserializeNewtonsoft | Job-UUWPJS | .NET Core 3.1 | 10000 | 177.367 ms | 364.9170 ms | 20.0023 ms |   II | 14000.0000 | 2000.0000 |        - |     87 MB |

**Comment by MartinCarpentier** [comment]
@matt-lethargic I'm working on setting up the continous benchmarking, but now it requests a github pages, where the result can be deployed.
I do not have the access rights for this. Can you set it up?
When done, it will also comment on pull requests if the performance have significantly degraded.

**Pull request merged**